src/stream/src/common/table/state_table.rs

```diff
@@ -177,7 +162,14 @@ impl<S: StateStore> StateTable<S> {
             .collect_vec();
 
         let dist_key_in_pk_indices = get_dist_key_in_pk_indices(&dist_key_indices, &pk_indices);
-        let local_state_store = store.new_local(table_id).await;
+        let table_option = TableOption::build_table_option(table_catalog.get_properties());
+        let local_state_store = store
+            .new_local(NewLocalOptions {
```

Comment:
minor: may rename `from_table_catalog_may_disable_sanity_check` to `from_table_catalog_inner`.